pkg/sdn/plugin/node.go

```diff
@@ -127,11 +139,138 @@ func (node *OsdnNode) Start() error {
 		}
 	}
 
+	if err := node.cniServer.Start(); err != nil {
+		return err
+	}
+	go utilwait.Forever(node.watchCni, 0)
+
 	node.markPodNetworkReady()
 
 	return nil
 }
 
+type podWorker struct {
+	node     *OsdnNode
+	requests []*podRequest
+	lock     sync.Mutex
+}
+
+func newPodWorker(node *OsdnNode) *podWorker {
+	return &podWorker{
+		node:     node,
+		requests: make([]*podRequest, 0),
+	}
+}
+
+func (w *podWorker) addRequest(request *podRequest) {
+	w.lock.Lock()
+	defer w.lock.Unlock()
+	w.requests = append(w.requests, request)
+	if len(w.requests) == 1 {
+		// Start proessing when the first request is added
+		go w.process()
+	}
+}
+
+func (w *podWorker) numRequests() int {
+	w.lock.Lock()
+	defer w.lock.Unlock()
+	return len(w.requests)
+}
+
+func (w *podWorker) process() {
+	for {
+		// Grab an item off the queue
+		w.lock.Lock()
+		if len(w.requests) == 0 {
+			w.lock.Unlock()
+			return
+		}
+		request := w.requests[0]
+		w.requests = w.requests[1:]
+		w.lock.Unlock()
```

Comment:
@smarterclayton requested this in https://github.com/openshift/origin/pull/11082#discussion_r80743524 if I understand his request correctly.  And he's at least right from the perspective that if pod setup goes into the weeds for a single pod, that would (in the original push) block any subsequent pods from being created.